### ISSUE TITLE
Immutables no longer generates code that compiles with Java 1.6

### ISSUE THREAD

**Issue by tjwilson90** [immutables/immutables#297]
Between 2.1.11 and 2.1.12 the code generation for formatRequiredAttributesMessage changed a line from
```
List<String> attributes = new ArrayList<String>();
```
to
```
List<String> attributes = new ArrayList<>();
```
which breaks support of Java 1.6, fairly unnecessarily.

**Comment by elucash** [comment]
My bad, sorry. This one will be fixed. We will not advertise java 6 support in general and some advanced processors/capabilities might not generate valid java 6. But for simple cases like this we need to retain the compatibility.

**Comment by tjwilson90** [comment]
Awesome, thanks.

As long as the feature set as of org.immutables 2.0 works in java 1.6, I'll be happy.

**Comment by elucash** [comment]
Just reviewed the source/templates. I have a question though. Do you use Guava? In general our Java 6 compatibility relies on the presence of Guava, in this case `Lists.newArrayList()` would be used instead of diamond constructor. I just worry that there are couple of other places where this might strike-back, and using Guava all will be easier on Java 6

**Comment by trask** [comment]
Thanks for continued Java 6 runtime support.  Using Guava seems like a reasonable requirement for Java 6 users (like me).

**Comment by tjwilson90** [comment]
We don't use guava (in the projects that I care about for this, mostly api projects that need to be broadly available, hence 1.6, and have minimal dependencies). Up until now, immutables has been generating 1.6 compatible code without guava (at least with the features I care about).

**Comment by elucash** [comment]
Ok, it makes sense to fix these kind of [simple] issues, without diving into advanced stuff where java7 or guava required.

**Comment by elucash** [comment]
Released in v 2.1.16 Thank you!
